default/python/universe_generation/monsters.py

```diff
@@ -1,8 +1,110 @@
+import sys
 import random
 import freeorion as fo
 import util
 import statistics
 import universe_tables
+from galaxy import DisjointSets
+
+
+def populate_monster_fleet(fleet_plan, system):
+    """
+    Create a monster fleet in ''system'' according to ''fleet_plan''
+    """
+
+    # create monster fleet
+    monster_fleet = fo.create_monster_fleet(system)
+    # if fleet creation fails, report an error and try to continue with next system
+    if monster_fleet == fo.invalid_object():
+        errstr = "Python generate_monsters: unable to create new monster fleet %s" % fleet_plan.name()
+        util.report_error(errstr)
+        raise RuntimeError(errstr)
```

Comment:
If you throw an exception, don't use `report_error` here (which will write to the error log), but on the level where you handle the exception. Currently this causes the error to be logged twice.